docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -544,3 +544,50 @@ PUT _cluster/settings
   }
 }
 ----
+
+[discrete]
+==== Number of documents in the index can't exceed [2147483519]
+
+Elasticsearch shards reflect Lucene's underlying https://github.com/apache/lucene/issues/5176[index 
+`MAX_DOC` hard limit] of 2,147,483,519 (`(2^31)-1`) docs. This figure calulates 
+from the sum of `docs.count` plus `docs.deleted` as reported by <<indices-stats,Index Stats>> 
+per shard. Attempting to surpass this limit, e.g. via ingestion's <<index-modules-translog,translog>>, 
+will cause error 
+
+[source, txt]
+----
+Elasticsearch exception [type=illegal_argument_exception, reason=Number of documents in the index can't exceed [2147483519]]
+----
+
+or this error's sister stack trace will appear 
+
+[source, txt]
+----
+Caused by: java.lang.IllegalArgumentException: number of documents in the index cannot exceed 2147483519
+----
+
+Noting that in this error message, `index` is referencing a Lucene index which correlates to an 
+<<glossary-shard,Elasticsearch shard>>. 
```

Comment:
This exception is actually thrown from elasticsearch. I wonder if we should change it to avoid confusion?

https://github.com/elastic/elasticsearch/blob/6ae652f90e7b15d9eef81fa745da299d39845655/server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java#L1691